### PR TITLE
TST: add tests for Series/DF logical operations nan propagation

### LINKED ISSUE #13896: API: Series/DataFrame logical op NaN propagation
The same as #13806, but for discuss differences with `DataFrame`.

```
pd.Series([True, False, np.nan]) | pd.Series([True, False, True])
#0     True
#1    False
#2    False
# dtype: bool

pd.Series([True, False, True]) | pd.Series([True, False, np.nan])
#0     True
#1    False
#2     True
# dtype: bool

pd.DataFrame([True, False, np.nan]) | pd.DataFrame([True, False, True])
#        0
#0   True
#1  False
#2    NaN

pd.DataFrame([True, False, True]) | pd.DataFrame([True, False, np.nan])
#        0
#0   True
#1  False
#2    NaN
```

There looks to be 2 points:
## 1. `NaN` comparison result

As @TomAugspurger described in #13806, the basic rule is below. Based on this, `DataFrame` result is incorrect.

```
In [40]: float('nan') or 1
Out[40]: nan

In [41]: 1 or float('nan')
Out[41]: 1
```
## 2. filling `NaN` result

`Series` fills `NaN` included in the result with `False`. `DataFrame` should also?

### PR DESCRIPTION
**Pull request by ganevgv**
- [x] closes #13896
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/test_operators.py [hunk 1]
```diff
@@ -218,6 +218,20 @@ def test_logical_with_nas(self):
         expected = Series([True, True])
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize(
+        "A, B, C",
```
Comment (mroeschke) [inline pandas/tests/frame/test_operators.py]: Can you name these `"left, right, expected"`?

Comment (ganevgv) [reply pandas/tests/frame/test_operators.py]: Thanks, @mroeschke - done.

**Review by mroeschke — APPROVED** [review]

File: pandas/tests/frame/test_operators.py [hunk 2]
```diff
@@ -218,6 +218,20 @@ def test_logical_with_nas(self):
...
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize(
+        "left, right, expected",
+        [
+            ([True, False, np.nan], [True, False, True], [True, False, False]),
+            ([True, False, True], [True, False, np.nan], [True, False, True]),
+        ],
+    )
+    def test_logical_operators_nans(self, left, right, expected):
```
Comment (jreback) [inline pandas/tests/frame/test_operators.py]: can you add tests for and as well

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/test_operators.py (hunk 2 above)
Comment (ganevgv) [reply pandas/tests/frame/test_operators.py]: Thanks for the feedback, @jreback - done.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ganevgv

**Pull request merged**